common/mails/courses/index.ts

```diff
@@ -203,23 +205,44 @@ async function getCourseStartDate(subcourseId: number) {
     return firstLecture;
 }
 
-export async function sendPupilCourseSuggestion(course: Course | Prisma.course, subcourse: Subcourse | Prisma.subcourse) {
+export async function sendPupilCourseSuggestion(course: Course | Prisma.course, subcourse: Subcourse | Prisma.subcourse, actionId: ActionID) {
     const minGrade = subcourse.minGrade;
     const maxGrade = subcourse.maxGrade;
     const courseStartDate = await getCourseStartDate(subcourse.id);
+    const courseCapacity = await getCourseCapacity(subcourse);
+    const { alreadyPromoted, publishedAt } = subcourse;
+    const courseSubject = course.subject;
     const grades = [];
 
+    if (!courseStartDate) {
+        throw new Error(`Cannot send course suggestion mail for subcourse with ID ${subcourse.id}, because that course has no specified first lecture`);
+    }
+
     for (let grade = minGrade; grade <= maxGrade; grade++) {
         grades.push(`${grade}. Klasse`);
     }
 
-    // TODO filter pupils
     const pupils = await prisma.pupil.findMany({
         where: { active: true, isParticipant: true, grade: { in: grades } },
     });
 
-    for (let pupil of pupils) {
-        await Notification.actionTaken(pupil, 'instructor_subcourse_published', {
+    const getDaysDifference = (date: string): number => {
+        const today = new Date().getDay();
+        const published = new Date(date).getDay();
```

Comment:
`Date.prototype.getDay` returns a number between 0 and 6 which represents the day of the week and can not be used to calculate the day difference because of the switch from saturday 6 to sunday 0.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay?retiredLocale=de